astropy/io/fits/fitsrec.py

```diff
@@ -1224,24 +1224,25 @@ def _scale_back_ascii(self, col_idx, input_field, output_field):
             lead = 0
 
         if lead < 0:
-            warnings.warn('Column %r starting point overlaps the previous '
-                          'column.' % (col_idx + 1))
+            warnings.warn('Column {!r} starting point overlaps the previous '
+                          'column.'.format(col_idx + 1))
 
         trail = starts[col_idx + 1] - starts[col_idx] - spans[col_idx]
 
         if trail < 0:
-            warnings.warn('Column %r ending point overlaps the next '
-                          'column.' % (col_idx + 1))
+            warnings.warn('Column {!r} ending point overlaps the next '
+                          'column.'.format(col_idx + 1))
 
         # TODO: It would be nice if these string column formatting
         # details were left to a specialized class, as is the case
         # with FormatX and FormatP
+        print(format)
         if 'A' in format:
-            _pc = '%-'
+            _pc = '{:'
         else:
-            _pc = '%'
+            _pc = '{:>'
 
-        fmt = ''.join([_pc, format[1:], ASCII2STR[format[0]],
+        fmt = ''.join([_pc, format[1:], ASCII2STR[format[0]], '}',
```

Comment:
nope, there are two cases, it as to be either left or right aligned and the line above takes care of it.